src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DefaultProjectSnapshot.cs

```diff
@@ -58,9 +59,37 @@ public override DocumentSnapshot GetDocument(string filePath)
             }
         }
 
+        public override bool IsImportDocument(DocumentSnapshot document)
+        {
+            if (document == null)
+            {
+                throw new ArgumentNullException(nameof(document));
+            }
+            
+            return State.ImportsToIncludingDocuments.ContainsKey(document.TargetPath);
```

Comment:
State now tracks the 'target path' of each import document and which 'related files' it affects. These this is the API on PS for interacting with it.